File: fe-pc/ui/src/main/java/com/example/ui/apis/PostApi.java
package com.example.ui.apis;

import static com.example.ui.constant.constant.BASE_URL;

public class PostApi {
    public static final String CREATEPOST_ENDPOINT = "/private/post/create-with-file";
    public static final String GETALLPOST_ENDPOINT = BASE_URL + "/public/post/f/all";
    public static final String GETPOSTBYID_ENDPOINT = BASE_URL + "/public/post";
    public static final String DELETEPOST_ENDPOINT = BASE_URL + "/private/post/delete";
    public static final String UPDATEPOST_ENDPOINT = BASE_URL + "/private/post/update";
    
    public static final String savePostApi = BASE_URL + "/private/post/save";
    public static final String getAllPostUserApi = BASE_URL + "/private/post/c/user";
    public static final String getAllSavePostApi = BASE_URL + "/private/post/d/saved";
    public static final String unavePostApi = BASE_URL + "/private/post/a/saved";
    public static final String getAllNotificationApi = BASE_URL + "/private/notification";

    public static final String createCommentPostApi = BASE_URL + "/private/comment/create";

    public static String createPostEndpoint() {
        return BASE_URL + CREATEPOST_ENDPOINT;
    }
}

File: fe-mobile/lifetipsui/app/src/main/java/com/example/lifetipsui/apis/PostApi.kt
package com.example.lifetipsui.apis
import com.example.lifetipsui.config.Config

object PostApi {
    const val createPostApi = "${Config.BASE_URL}/private/post/create-with-file"
    const val savePostApi = "${Config.BASE_URL}/private/post/save"
    const val unsavePostApi = "${Config.BASE_URL}/private/post/a/saved"
    const val getAllByUserPostApi = "${Config.BASE_URL}/private/post/c/user"
    const val getAllSavePostApi = "${Config.BASE_URL}/private/post/d/saved"
    const val updatePostApi = "${Config.BASE_URL}/private/post/update"
    const val deletePostApi = "${Config.BASE_URL}/private/post/delete"


    const val searchPostApi = "${Config.BASE_URL}/public/post/e/search"
    const val getByIdPostApi = "${Config.BASE_URL}/public/post"
    const val getRandomPostApi = "${Config.BASE_URL}/public/post/a/random"
    const val getSameCategoryPostApi = "${Config.BASE_URL}/public/post/b/same-type/"
}